### PR TITLE
LPS-135308 Upgrade `recharts` version

### PR DESCRIPTION
**Pull request by kresimir-coko**
[Jira ticket](https://liferay.atlassian.net/browse/LPS-135308)

Running `yarn why d3-color` inside `modules` returns a mention of version `1.2.3` which is due to `frontend-taglib-chart`'s dependency on `clay-charts` - I left my thoughts and conclusion in the Jira ticket comment

### CODE REVIEW

**Review by markocikos — APPROVED** [review]
@kresimir-coko LGTM! :sparkles:

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by deborasoliveira** [comment]
I'm taking a look at the relevant and stable results.

**Comment by deborasoliveira** [comment]
There are some error messages similar to the one below in the `js-unit-jdk8/0/0` Jenkins report, could you take a look at it @kresimir-coko ? 
```
     [exec] Summary of all failing tests
     [exec] FAIL test/js/custom/form-report/components/chart/pie/PieChart.js
     [exec]   ??? PieChart ??? renders
     [exec] 
     [exec]     expect(received).toMatchSnapshot()
     [exec] 
     [exec]     Snapshot name: `PieChart renders 1`
     [exec] 
     [exec]     - Snapshot  - 1
     [exec]     + Received  + 1
     [exec] 
     [exec]     @@ -3,11 +3,11 @@
     [exec]           class="lfr-de-recharts pie-chart"
     [exec]           style="height: 100%; width: 100%;"
     [exec]         >
     [exec]           <div
     [exec]             class="recharts-responsive-container"
     [exec]     -       style="width: 700px; height: 300px;"
     [exec]     +       style="width: 700px; height: 300px; min-width: 0;"
     [exec]           >
     [exec]             <div
     [exec]               class="recharts-wrapper"
     [exec]               role="region"
     [exec]               style="position: relative; cursor: default; width: 700px; height: 300px;"
     [exec] 
     [exec]       40 | 		const {asFragment} = render(<PieChart {...props} />);
     [exec]       41 | 
     [exec]     > 42 | 		expect(asFragment()).toMatchSnapshot();
     [exec]          | 		                     ^
     [exec]       43 | 	});
     [exec]       44 | 
     [exec]       45 | 	it('expands a sector with the percentage label when mouse is over', () => {
     [exec] 
     [exec]       at Object.<anonymous> (test/js/custom/form-report/components/chart/pie/PieChart.js:42:24)
     [exec]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

The other failures seem to be unrelated. Many integration tests are failing because of:

[LPS-195242](https://liferay.atlassian.net/browse/LPS-195242)
[LPS-194047](https://liferay.atlassian.net/browse/LPS-194047)

**Comment by kresimir-coko** [comment]
ci:test:relevant

File: modules/apps/data-engine/data-engine-js-components-web/test/js/__mock__/MetalFieldMock.soy.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/* jshint ignore:start */
+import Component from 'metal-component';
```
Comment (markocikos) [inline modules/apps/data-engine/data-engine-js-components-web/test/js/__mock__/MetalFieldMock.soy.js:2]: This looks weird. We do have these types of artifacts in other places in DXP, but I wouldn't expect a new dependency version to force us into creating new artifacts.

Comment (kresimir-coko) [reply modules/apps/data-engine/data-engine-js-components-web/test/js/__mock__/MetalFieldMock.soy.js:2]: Yeah, I'm flabbergasted myself, I wanted to try to see if CI will pass with the changes, and the changes do look weird but locally `yarn test` passes...

Comment (kresimir-coko) [reply modules/apps/data-engine/data-engine-js-components-web/test/js/__mock__/MetalFieldMock.soy.js:2]: I might be onto something...

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:forward

**Comment by kresimir-coko** [comment]
ci:reevaluate:1337401_4487

**Comment by kresimir-coko** [comment]
ci:reevaluate:1337401_4487

**Comment by deborasoliveira** [comment]
ci:test:relevant

**Comment by deborasoliveira** [comment]
I'm checking the results.

**Comment by deborasoliveira** [comment]
The failures seem to be unrelated.

[Comparison run](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=2&testrayRunIdB=2441285044&testrayRunIdA=2440865274&statusB=3)

**Comment by kresimir-coko** [comment]
Manually forwarded to https://github.com/brianchandotcom/liferay-portal/pull/142265

**Pull request closed without merging**